src/disco/components/InstallButton.js

```diff
@@ -41,6 +43,28 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  getLabel() {
+    const { i18n, name, status, type } = this.props;
+    if (status === DOWNLOADING) {
+      return i18n.sprintf(i18n.gettext('Downloading %(name)s'), { name });
+    } else if (status === INSTALLING) {
+      return i18n.sprintf(i18n.gettext('Installing %(name)s'), { name });
+    } else if ([ENABLED, INSTALLED].includes(status) && type === THEME_TYPE) {
+      return i18n.sprintf(
+        i18n.gettext('%(name)s is installed and enabled. Click to disable'), { name });
+    } else if ([ENABLED, INSTALLED].includes(status) && type === EXTENSION_TYPE) {
+      return i18n.sprintf(
+        i18n.gettext('%(name)s is installed and enabled. Click to uninstall'), { name });
+    } else if (status === DISABLED) {
+      return i18n.sprintf(i18n.gettext('%(name)s is disabled. Click to enable'), { name });
+    } else if (status === UNINSTALLING) {
+      return i18n.sprintf(i18n.gettext('Uninstalling %(name)s'), { name });
+    } else if (status === UNINSTALLED) {
+      return i18n.sprintf(i18n.gettext('%(name)s is uninstalled. Click to install'), { name });
+    }
+    return i18n.gettext('State is unknown');
```

Comment:
This is the only one that doesn't have `name` in it. Maybe it should be part of the string (like "Install status of %(name)s is unknown.") and then the `sprintf` bit could happen once at the end.